server/models/web_vtt.py

```diff
@@ -0,0 +1,250 @@
+from pydantic import BaseModel, field_validator
+from typing import List, Literal, Dict, Optional
+from transcriber import Transcript
+
+import re
+
+# any integer or percentage between 0% and 100% inclusive:
+line_regex = re.compile(r"^-?\d+$|^100%$|^[1-9]?[0-9]%$")
+# any percentage between 0% and 100% inclusive:
+clamped_percent_regex = re.compile(r"^100%$|^[1-9]?[0-9]%")
+# css class / id name
+css_class_id_regex = re.compile(r"^[A-Za-z_].*$")
+
+# reference: https://www.tutorialrepublic.com/css-reference/css3-properties.php
+CSS_PROPERTIES = [
```

Comment:
The styling is being handled by the frontend all you had to do is just generate the vtt and send it to the frontend. We would worry about the cue styling

Reply:
The styling here is completely optional, and is only there for future extensibility :) .

You can style the captions on the frontend, fine, but the client is may want the ability to download custom WebVTT captions in the future.